### DISCUSSION TITLE
Quais são os problemas de trabalhar em agências?

### DISCUSSION THREAD

**Discussion by Murillofilho86** [frontendbr/forum#1087] [Discussão]
Estou migrando de back-end para front-end e vejo muitas vagas em agências, porém nunca trabalhei em nenhuma, mas sempre vi um comentário ou outro de que trabalhar em agência era ruim, afinal, qual é o problema das agências?

**Comment by tarciozemel** [comment]
O mesmo que de fora das agências: prazos beirando o impossível com exigência de ótima qualidade; horas-extras sem recompensa; trabalho em fim de semana; escopo de atuação fora do combinado; etc.

**Comment by kvnol** [comment]
> Quais são os problemas de trabalhar em agências?

Trabalhar com publicitários.

**Comment by Murillofilho86** [comment]
@tarciozemel isso quer dizer que não tem nenhum problema? 🤔 Kkkkk

**Comment by bernardodiasc** [comment]
O problema maior eh ter que trabalhar pra pessoas que não entendem o estão fazendo.

**Comment by renanlara** [comment]
@Murillofilho86, posso estar errado, mas não vá por mim. Se fosse o seu o meu caso, eu não faria isso. Continuaria no back-end, me aprofundaria mais se possível e aprenderia só o necessário de front.

Trabalhar em agência já é uma loucura e, sendo fron-end se torna pior. Um exemplo disso é o que a galera anda discutindo aqui #1077. O que o @tarciozemel comentou é apenas um breve resumo.

**Comment by felipefialho** [comment]
Trabalhei bastante tempo em agências, com experiencias bem distintas.

- quase 4 anos numa agência brasileira (muito tempo, zero retorno, erro meu total). 
- 2 anos remoto em uma agência gringa (aqui a experiencia foi ótima).

### Pontos positivos 

- Se a agência for boa, vai melhorar seu senso estético
- Se a agência for boa², melhorar o portfólio com projetos esteticamente bonitos
- Possibilidade de começar projetos do 0 
- Horário tende a ser flexível (mas não era na agência do Brasil que eu trabalhei, por exemplo)

### Pontos negativos

- Pastéis
- Ritmo de trabalho enlouquecedor
- Pouca profundidade nos projetos
- Legados
- Suporte para browsers antigos
- Na média paga abaixo (as vezes bem abaixo) do mercado

**Comment by Cleysonlb** [comment]
Trabalhei por um bom tempo em agências também, é exatamente o que o @LFeh falou!
Na minha experiência, outro ponto que me incomodava era a estagnação profissional, a evolução era mínima, usando sempre as mesmas tecnologias e processos de desenvolvimento, raramente recebia um feedback sobre o trabalho e pouco senso critico sobre o job, já que era uma esteira de produção tipo fábrica, e uma rotina intensa com prazos apertados.
Mas... Dando outra visão sobre, não é de todo mal, aprendi muito, tive oportunidade de incrementar portfólio, realizar projetos bacanas para diversas marcas/empresas, e até aconselho a experiência contando que não seja longa. Espero ter ajudado!

**Comment by angelosds** [comment]
Trabalho em agência atualmente, na realidade, é o primeiro trabalho na área e já são quase 5 anos em uma das agências digitais mais tradicionais (24 anos no mercado) de São Paulo, e pretendo sair daqui em breve.

Assim como o @LFeh disse, há pontos positivos e negativos, talvez o maior ponto positivo seja a possibilidade de trabalhar com diversos projetos diferentes e participar de todo o processo de desenvolvimento, além de flexibilidade de horários e liberdade de usar a roupa que você quiser.

Porém, na minha experiência, há mais pontos negativos e eles pesam mais. 

1. O salário é bastante abaixo do mercado.
2. Os prazos são na maioria dos casos, curtos.
3. Os profissionais tendem a não evoluir (em questão de ferramentas e tecnologias) por conta de suporte à navegadores antigos e falta de tempo para estudar.

Vale ressaltar, que essa é a experiência que tenho tido no meu trampo atual, não necessariamente a realidade de todas as agências. Boa sorte cara!

**Comment by cezarsmpio** [comment]
No começo da minha carreira eu trabalhei em 3 agências até que comecei a trabalhar em empresas mais voltadas a produto/software.

Das 3, tive problema apenas com 1.

Nas outras acho que foram os lugares onde eu mais aprendi, bom era o início da minha carreira também então tudo era novidade.

Lembro que aprendi muito de tudo, desde front, back, banco de dados, wordpress pois tínhamos que criar projetos sempre do zero, então eu testava diferentes tecnologias, frameworks, libs, isso foi incrível. Cada projeto era uma stack diferente. Trabalho dos sonhos haha

Por outro lado, tínhamos alguns problemas com os designers pois muitos não eram web designers e sim designers gráficos, então alguns projetos eram bem complicados, tanto de comunicação e entendimento do que o projeto precisava. Isso começou a ser resolvido quando começamos a contratar pessoas voltadas para web e o setor ficou mais independente.

Acho que realmente vai de agência pra agência. Em uma delas eu trabalhei por quase 4 anos e aprendi muito pois essa agência me permitia isso.

Existem vários tipos de agências, tem que levar isso em consideração também. Tem muitas agências que trabalham com varejo e eu não recomendo essas por experiência própria. Existem outras onde os projetos são maiores e trabalham com clientes grandes o que ajuda no seu portfólio.

Um dos problemas mais tensos era o prazo, onde o design consumia 90% dele, por exemplo, o projeto tem 1 mês, 3 semanas era apenas do design enquanto você tinha 1 semana para fazer tudo. Isso também vai de como a empresa é organizada, mas geralmente agências não contam product owners e acho que isso sempre tende a acontecer.

Os salários tendem a ser mais baixos também, isso é verdade. 

Bom, fica o meu relato aqui. Veja quais são as suas prioridades nesse momento ;)

**Comment by Morette** [comment]
Pessoal, desculpa a pergunta idiota, mas qual a diferença entre agência e empresa?

Estou tentando entrar mudar de área (hoje trabalho empresa no ramo de petróleo) e vendo essa issue, não consegui entender a diferença..
😅

**Comment by bernardodiasc** [comment]
@Morette agencias no caso, sao geralmente agencias de publicidade, elas atendem demandas de outros clientes e produzem ações de propaganda. Pra quem trabalha com desenvolvimento web isso significa fazer muitos sites pequenos, institucionais, as vezes alguns maiores e tal, mas comumente prazos apertados e os colegas de trabalho e chefes costumam ser de uma outra área tornando o ambiente mais dificil para o profissional técnico. No caso de empresas, geralmente se tratam de demandas maiores, produtos que tem uma vida mais longa, colegas de trabalhos e gerencia que entende o que é o trabalho de programação e etc.

Na minha opinião o problema de agencias se resume a gestão que não entende o trabalho do técnico. Tive experiencias extremamente desafiadoras trabalhando em agencias, evolui muito, mas foi na marra, na marretada. Quando fui trabalhar em uma empresa onde as pessoas eram da mesma área que eu, era tudo diferente, desde processos até o tratamento das pessoas.
